components/Crypto.tsx

```diff
@@ -0,0 +1,194 @@
+import React, {useEffect, useRef} from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  Animated,
+  Image,
+  TouchableOpacity,
+} from 'react-native';
+
+import {useNavigation} from '@react-navigation/native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import {Crypto} from '../models/types';
+import AnimatedText from './AnimatedText';
+import Colors from '../utils/Colors';
+
+interface Props {
+  crypto: Crypto;
+}
+
+function CryptoDetail({crypto}: Props) {
+  const navigation = useNavigation();
+  const spinAnim = useRef(new Animated.Value(0)).current;
+
+  useEffect(() => {
+    Animated.loop(
+      Animated.timing(spinAnim, {
+        toValue: 1,
+        duration: 1200,
+        useNativeDriver: true,
+      }),
+    ).start();
+  }, [spinAnim]);
+
+  const spin = spinAnim.interpolate({
+    inputRange: [0, 1],
+    outputRange: ['0deg', '360deg'],
+  });
+
+  const handlePress = () => {
+    if (navigation)
+      navigation.navigate('DetailCrypto' as never, {crypto: crypto} as never);
+  };
+
+  var value;
+  var valuePercentage;
+
+  if (crypto.market_data.price_usd) {
+    value = crypto.market_data.price_usd.toLocaleString('en-US', {
```

Comment:
in this case you should instead of a state or a variable, use a useMemo, so you can store it, and prevent unnecessary re declarations or re renders:

```
const value = useMemo(() => {
   // put the validation here
   return crypto.market_data.price_usd.toLocaleString......
}, [crypto.market_data.price_usd])
```